package/MDAnalysis/converters/RDKitInferring.py

```diff
@@ -0,0 +1,703 @@
+        +---------------+------------------------------------------------------------------------------+
+        | keto-enolate  | ``[#6-:1]-[#6:2]=[O:3]>>[#6+0:1]=[#6:2]-[O-:3]``                             |
+        +---------------+------------------------------------------------------------------------------+
+        | arginine      | ``[C-;v3:1]-[#7+0;v3;H2:2]>>[#6+0:1]=[#7+:2]``                               |
+        +---------------+------------------------------------------------------------------------------+
+        | histidine     | ``[#6+0;H0:1]=[#6+0:2]-[#7;X3:3]-[#6-;X3:4]>>[#6:1]=[#6:2]-[#7+:3]=[#6+0:4]``|
+        +---------------+------------------------------------------------------------------------------+
+        | sulfone       | ``[S;D4;!v6:1]-[*-:2]>>[S;v6:1]=[*+0:2]``                                    |
+        +---------------+------------------------------------------------------------------------------+
+        | charged N     | ``[#7+0;X3:1]-[*-:2]>>[#7+:1]=[*+0:2]``                                      |
+        +---------------+------------------------------------------------------------------------------+
+
+        """
+        if max_iter is None:
+            max_iter = self.max_iter
+        else:
+            warnings.warn(
+                "Specifying `max_iter` is deprecated and will be removed in a "
+                "future update. Directly create an instance of "
+                "`MDAnalysisInferer` with `MDAnalysisInferer(max_iter=...)` "
+                "instead.",
+                DeprecationWarning,
+            )
+
+        # standardize conjugated systems
+        self._rebuild_conjugated_bonds(mol, max_iter)
+
+        # list of sanitized reactions
+        reactions = [
+            ReactionFromSmarts(rxn) for rxn in self.STANDARDIZATION_REACTIONS
+        ]
+
+        # fragment mol (reactions must have single reactant and product)
+        fragments = list(Chem.GetMolFrags(mol, asMols=True))
+        for reactant in fragments:
+            self._apply_reactions(reactions, reactant)
+
+        # recombine fragments
+        newmol = fragments.pop(0)
+        for fragment in fragments:
+            newmol = Chem.CombineMols(newmol, fragment)
+
+        return newmol
+
+    @staticmethod
+    def _apply_reactions(
+        reactions: List["ChemicalReaction"], reactant: "Chem.Mol"
+    ) -> None:
+        """Applies a series of unimolecular reactions to a molecule. The
+        reactions must not add any atom to the product. The molecule is
+        modified inplace.
+
+        Parameters
+        ----------
+        reactions : List[rdkit.Chem.rdChemReactions.ChemicalReaction]
+            Reactions from SMARTS. Each reaction is applied until no more
+            transformations can be made.
+        reactant : rdkit.Chem.rdchem.Mol
+            The molecule to transform inplace
+
+        """
+        reactant.UpdatePropertyCache(strict=False)
+        Chem.Kekulize(reactant)
```

Comment:
note to self: use `KekulizeIfPossible` instead, same below and in _rebuild_conjugated_bonds